lib/internal/process/promises.js

```diff
@@ -99,3 +99,12 @@ function setupPromises(scheduleMicrotasks) {
 
   return emitPendingUnhandledRejections;
 }
+
+const savedString = String;
+function safeString(value) {
+  try {
+    return savedString(value);
+  } catch (err) {
+    return '<toString() threw exception>';
```

Comment:
I'm ok for crashing badly in that case. It's always better to crash in those situations.